batch/batch/resource_usage.py

```diff
@@ -212,6 +220,11 @@ async def periodically_measure():
                 except asyncio.CancelledError:
                     cancelled = True
                     raise
+                except OSError as err:
+                    if err.errno == errno.ENOSPC:
+                        cancelled = True
```

Comment:
I think this will kill the user's job. I'm not certain if we want that behavior or just to cancel the monitoring loop.

Reply:
Why does this kill the user's job? Since this is happening in a separate task, I don't see anywhere where this exception is handled.